ca_tools/scripts/path_printer.py

```diff
@@ -0,0 +1,47 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+import math
+
+import rospy
+from GTSys import GTSys
+from geometry_msgs.msg import Pose, PoseStamped
+from nav_msgs.msg import Path
+
+class pathPrinter():
+
+    def __init__(self):
+        self._path = Path()
+        self._path_pub = rospy.Publisher('path', Path, queue_size=10)
+        self._gt = GTSys()
+
+        self._rate = rospy.Rate(1) #1 Path msg per second
+        self._rate.sleep()
+
+    def set_path_frame_id(self,frame):
+        self._path.header.frame_id = frame
+        self._gt.set_frame_id(frame)
+
+    def _update_path(self):
+        self._current_header = self._gt.get_header()
+        self._current_pose = self._gt.get_pose()
+        pose = PoseStamped()
+        pose.header = self._gt.get_header()
+        pose.pose = self._gt.get_pose()
+        self._path.poses.append(pose)
+
+
+    def run(self):
+        while True:
+            self._update_path()
+            self._path_pub.publish(self._path)
+            self._rate.sleep()
+
+def main():
+    rospy.init_node('path_printer')
+    app = pathPrinter()
+    app.set_path_frame_id("/map")
```

Comment:
Why don't you set it in the constructor?

Reply:
I wanted to make the frame_id a parameter of the class. If I'll want to reuse the code in the future, I wanted it to be made in such a way that I won't have to modify the working code.
I wanted initially to call the frame_id when I create the instance, but I hadn't had time to check how to do it. I'll investigate about that.